tests/test_tsdownsample.py

```diff
@@ -9,99 +11,81 @@
     MinMaxDownsampler,
     MinMaxLTTBDownsampler,
 )
+from tsdownsample.downsampling_interface import AbstractDownsampler
 
 # TODO: Improve tests
 #   - compare implementations with existing plotly_resampler implementations
 
 
-def test_m4_downsampler():
-    """Test M4 downsampler."""
-    arr = np.array(np.arange(10_000))
-    s_downsampled = M4Downsampler().downsample(arr, n_out=100)
-    assert s_downsampled[0] == 0
-    assert s_downsampled[-1] == len(arr) - 1
+RUST_DOWNSAMPLERS = [
+    MinMaxDownsampler(),
+    M4Downsampler(),
+    LTTBDownsampler(),
+    MinMaxLTTBDownsampler(),
+]
 
+OTHER_DOWNSAMPLERS = [EveryNthDownsampler()]
 
-def test_minmax_downsampler():
-    """Test MinMax downsampler."""
-    arr = np.array(np.arange(10_000))
-    s_downsampled = MinMaxDownsampler().downsample(arr, n_out=100)
-    assert s_downsampled[0] == 0
-    assert s_downsampled[-1] == len(arr) - 1
 
+def generate_rust_downsamplers() -> Iterable[AbstractDownsampler]:
+    for downsampler in RUST_DOWNSAMPLERS:
+        yield downsampler
 
-def test_lttb_downsampler():
-    """Test LTTB downsampler."""
-    arr = np.array(np.arange(10_000))
-    s_downsampled = LTTBDownsampler().downsample(arr, n_out=100)
-    assert s_downsampled[0] == 0
-    assert s_downsampled[-1] == len(arr) - 1
+
+def generate_all_downsamplers() -> Iterable[AbstractDownsampler]:
+    for downsampler in RUST_DOWNSAMPLERS + OTHER_DOWNSAMPLERS:
+        yield downsampler
 
 
-def test_minmaxlttb_downsampler():
-    """Test MinMaxLTTB downsampler."""
-    arr = np.array(np.arange(10_000))
-    s_downsampled = MinMaxLTTBDownsampler().downsample(arr, n_out=100)
+@pytest.mark.parametrize("downsampler", generate_rust_downsamplers())
+def test_rust_downsampler(downsampler: AbstractDownsampler):
+    """Test the Rust downsamplers."""
+    arr = np.arange(10_000)
+    s_downsampled = downsampler.downsample(arr, n_out=100)
     assert s_downsampled[0] == 0
     assert s_downsampled[-1] == len(arr) - 1
 
 
 def test_everynth_downsampler():
     """Test EveryNth downsampler."""
-    arr = np.array(np.arange(10_000))
-    s_downsampled = EveryNthDownsampler().downsample(arr, n_out=100)
+    arr = np.arange(10_000)
+    downsampler = EveryNthDownsampler()
+    s_downsampled = downsampler.downsample(arr, n_out=100)
     assert s_downsampled[0] == 0
     assert s_downsampled[-1] == 9_900
```

Comment:
Having a seperate test is imo cleaner than having an if-case in the `test_rust_downsampler` (besides that, EveryNth is also not a rust downsampler)